### PR TITLE
New autoscaling lifecycle hook sdk

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
As the issue `Add Lifecycle hook resource suppoort for AS` of the Huaweicloud, we need to support related hook sdk.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 
```
NONE
```

**Special notes for your reviewer**:

**Release note**:

```release-note
1. new lifecycle hook sdk supported:
  - Create
  - Get
  - List
  - Update
  - Delete
2. add test for lifecycle hook sdk.
```

## Acceptance Steps Performed
```
go test -v -run Test
=== RUN   TestCreateLifecycleHook
--- PASS: TestCreateLifecycleHook (0.00s)
=== RUN   TestGetLifecycleHook
--- PASS: TestGetLifecycleHook (0.00s)
=== RUN   TestListLifecycleHook
--- PASS: TestListLifecycleHook (0.00s)
=== RUN   TestUpdateLifecycleHook
--- PASS: TestUpdateLifecycleHook (0.00s)
=== RUN   TestDeleteLifecycleHook
--- PASS: TestDeleteLifecycleHook (0.01s)
PASS
ok      github.com/huaweicloud/golangsdk/openstack/autoscaling/v1/lifecyclehooks/testing        0.028s
```

### CODE REVIEW

File: openstack/autoscaling/v1/lifecyclehooks/requests.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	_, r.Err = client.Put(resourceURL(client, groupID, hookName), b, &r.Body, &golangsdk.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+//Delete is a method which can be able to access to delete the existing hook of the autoscaling service.
+func Delete(client *golangsdk.ServiceClient, groupID, hookName string) (r NoneResult) {
+	_, r.Err = client.Delete(resourceURL(client, groupID, hookName), &golangsdk.RequestOpts{})
```
Comment (ShiChangkuo) [inline openstack/autoscaling/v1/lifecyclehooks/requests.go:106]: ditto.

File: openstack/autoscaling/v1/lifecyclehooks/results.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		// An array of lifecycle hooks.
+		Hooks []Hook `json:"lifecycle_hooks"`
+	}
+	err := r.Result.ExtractInto(&s)
+	return &s.Hooks, err
+}
+
+type NoneResult struct {
+	commonResult
+}
```
Comment (ShiChangkuo) [inline openstack/autoscaling/v1/lifecyclehooks/results.go:60]: rename the struct to `DeleteResult`, and use `golangsdk.ErrResult` as a member

File: openstack/autoscaling/v1/lifecyclehooks/requests.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	_, r.Err = client.Post(rootURL(client, groupID), b, &r.Body, &golangsdk.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// Get is a method to obtains the hook detail of autoscaling service.
+func Get(client *golangsdk.ServiceClient, groupID, hookName string) (r GetResult) {
+	_, r.Err = client.Get(resourceURL(client, groupID, hookName), &r.Body, &golangsdk.RequestOpts{})
```
[the remark this answers is not in the document]
Comment (Lance52259) [reply openstack/autoscaling/v1/lifecyclehooks/requests.go]: The *RequestOpts has updated with nil.

File: openstack/autoscaling/v1/lifecyclehooks/requests.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	return
+}
+
+// List is a method to obtains a hook array of the autoscaling service.
+func List(client *golangsdk.ServiceClient, groupID string) (r ListResult) {
+	_, r.Err = client.Get(listURL(client, groupID), &r.Body, &golangsdk.RequestOpts{})
```
[the remark this answers is not in the document]
Comment (Lance52259) [reply openstack/autoscaling/v1/lifecyclehooks/requests.go]: Done.

File: openstack/autoscaling/v1/lifecyclehooks/requests.go (hunk 1 above)
Comment (Lance52259) [reply openstack/autoscaling/v1/lifecyclehooks/requests.go]: Done.

File: openstack/autoscaling/v1/lifecyclehooks/results.go (hunk 2 above)
Comment (Lance52259) [reply openstack/autoscaling/v1/lifecyclehooks/results.go]: The struct name has been udpated to `DeleteResult`.

**Pull request merged**